lib/teiserver/lobby/libs/lobby_restrictions.ex

```diff
@@ -0,0 +1,282 @@
+        nil
+
+      # Just a max rating
+      consul_state.maximum_rank_to_play < @rank_upper_bound &&
+          consul_state.minimum_rank_to_play <= 0 ->
+        "Max chev: #{consul_state.maximum_rank_to_play + 1}"
+
+      # Just a min rating
+      consul_state.maximum_rank_to_play >= @rank_upper_bound &&
+          consul_state.minimum_rank_to_play > 0 ->
+        "Min chev: #{consul_state.minimum_rank_to_play + 1}"
+
+      # Chev range
+      # It shouldn't go here
+      consul_state.maximum_rank_to_play < @rank_upper_bound ||
+          consul_state.minimum_rank_to_play > 0 ->
+        "Chev between: #{consul_state.minimum_rank_to_play} - #{consul_state.maximum_rank_to_play}"
+
+      true ->
+        nil
+    end
+  end
+
+  def get_rating_bounds_for_title(consul_state) when consul_state == nil do
+    nil
+  end
+
+  def get_rating_bounds_for_title(consul_state) do
+    default_state = %{maximum_rating_to_play: @rating_upper_bound, minimum_rating_to_play: 0}
+    consul_state = Map.merge(default_state, consul_state)
+
+    cond do
+      # Default ratings
+      consul_state.maximum_rating_to_play >= @rating_upper_bound &&
+          consul_state.minimum_rating_to_play <= 0 ->
+        nil
+
+      # Just a max rating
+      consul_state.maximum_rating_to_play < @rating_upper_bound &&
+          consul_state.minimum_rating_to_play <= 0 ->
+        "Max rating: #{consul_state.maximum_rating_to_play}"
+
+      # Just a min rating
+      consul_state.maximum_rating_to_play >= @rating_upper_bound &&
+          consul_state.minimum_rating_to_play > 0 ->
+        "Min rating: #{consul_state.minimum_rating_to_play}"
+
+      # Rating range
+      consul_state.maximum_rating_to_play < @rating_upper_bound ||
+          consul_state.minimum_rating_to_play > 0 ->
+        "Rating between: #{consul_state.minimum_rating_to_play} - #{consul_state.maximum_rating_to_play}"
+
+      true ->
+        nil
+    end
+  end
+
+  def get_failed_rank_check_text(player_rank, consul_state) do
+    bounds = get_rank_bounds_for_title(consul_state)
+
+    [
+      @splitter,
+      "You don't meet the chevron requirements for this lobby (#{bounds}). Your chevron level is #{player_rank + 1}. Learn more about chevrons here:",
+      "https://www.beyondallreason.info/guide/rating-and-lobby-balance#rank-icons"
+    ]
+  end
+
+  def get_failed_rating_check_text(player_rating, consul_state, rating_type) do
+    bounds = get_rating_bounds_for_title(consul_state)
+    player_rating_text = player_rating |> Decimal.from_float() |> Decimal.round(2)
+
+    [
+      @splitter,
+      "You don't meet the rating requirements for this lobby (#{bounds}). Your #{rating_type} match rating is #{player_rating_text}. Learn more about rating here:",
+      "https://www.beyondallreason.info/guide/rating-and-lobby-balance#openskill"
+    ]
+  end
+
+  @doc """
+  Returns {:ok, nil} or {:error,msg}
+  """
+  def check_rank_to_play(user, consul_state) do
```

Comment:
Prefer using a typespec if you document the return type.
```
@spec check_rank_to_play(any(), any()) :: {:ok, nil} | {:error, String.t()}
```

Also, this is janky, `{:ok, nil}` should just be `:ok`, the nil is useless here.